test/itemlist_formaction.cpp

```diff
@@ -675,32 +616,39 @@ TEST_CASE("Navigate back and forth using OP_NEXT and OP_PREVIOUS", "[itemlist_fo
 	REQUIRE(line == prefix_title + first_article_title);
 }
 
-TEST_CASE("process_op(OP_RANDOMUNREAD)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+TEST_CASE("OP_TOGGLESHOWREAD switches the value of show-read-articles", "[itemlist_formaction]") {
+	controller c;
+	regexmanager regman;
+	newsbeuter::view v(&c);
+	configcontainer * cfg = c.get_cfg();
+	cache rsscache(":memory:", cfg);
 
-TEST_CASE("process_op(OP_NEXTUNREADFEED)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+	keymap k(KM_NEWSBEUTER);
+	v.set_keymap(&k);
 
-TEST_CASE("process_op(OP_PREVUNREADFEED)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+	v.set_regexmanager(&regman);
+	v.set_config_container(cfg);
+	c.set_view(&v);
 
-TEST_CASE("process_op(OP_NEXTFEED)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+	std::shared_ptr<rss_feed> feed = std::make_shared<rss_feed>(&rsscache);
 
-TEST_CASE("process_op(OP_PREVFEED)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+	std::shared_ptr<rss_item> item = std::make_shared<rss_item>(&rsscache);
+	feed->add_item(item);
 
-TEST_CASE("process_op(OP_MARKFEEDREAD)", "[itemlist_formaction]") {
-	//NOTIMPL
-}
+	itemlist_formaction itemlist(&v, itemlist_str);
+	itemlist.set_feed(feed);
+	v.push_itemlist(feed);
 
-TEST_CASE("process_op(OP_TOGGLESHOWREAD)", "[itemlist_formaction]") {
-	//NOTIMPL
+	SECTION("True to False"){
+		v.get_cfg()->set_configvalue("show-read-articles", "yes");
+		REQUIRE_NOTHROW(itemlist.process_op(OP_TOGGLESHOWREAD));
+		REQUIRE(!v.get_cfg()->get_configvalue_as_bool("show-read-articles"));
```

Comment:
Catch has `REQUIRE_FALSE` macro which I like because it's much more visible than negation operator. Please use it here.